File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/java/src/main/java/de/sots/cellarsandcentaurs/domain/service/CreatureService.java
package de.sots.cellarsandcentaurs.domain.service;

import de.sots.cellarsandcentaurs.domain.model.Creature;

public class CreatureService {
    private final Creatures creatures;

    public CreatureService(Creatures creatures) {
        this.creatures = creatures;
    }

    public void save(Creature creature) {
        creatures.save(creature);
    }
}

File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/csharp/src/de/sots/cellarsandcentaurs/domain/service/CreatureService.cs
using de.sots.cellarsandcentaurs.domain.model;

namespace de.sots.cellarsandcentaurs.domain.service
{
    [Service]
    public class CreatureService
    {
        private readonly Creatures creatures;

        public CreatureService(Creatures creatures)
        {
            this.creatures = creatures;
        }

        public void Save(Creature creature)
        {
            creatures.Save(creature);
        }
    }
}
